compiler/runtime/OMRCodeCacheManager.cpp

```diff
@@ -254,15 +254,18 @@ OMR::CodeCacheManager::freeMemory(void *memoryToFree)
 
 TR::CodeCache *
 OMR::CodeCacheManager::allocateCodeCacheObject(TR::CodeCacheMemorySegment *codeCacheSegment,
-                                             size_t codeCacheSizeAllocated,
-                                             CodeCacheHashEntrySlab *hashEntrySlab)
+                                               size_t codeCacheSizeAllocated)
    {
    TR::CodeCache *codeCache = static_cast<TR::CodeCache *>(self()->getMemory(sizeof(TR::CodeCache)));
    if (codeCache)
       {
       new (codeCache) TR::CodeCache();
-      if (!codeCache->initialize(self(), codeCacheSegment, codeCacheSizeAllocated, hashEntrySlab))
+      if (!codeCache->initialize(self(), codeCacheSegment, codeCacheSizeAllocated))
          {
+         if (codeCache->hashEntrySlab())
+            {
+            codeCache->hashEntrySlab()->free(self());
+            }
```

Comment:
Now that the `CodeCache` object allocates the slab itself it seems like a bad idea for the caller to have to clean it up. Can't this be done in `initialize()` if it fails?

Reply:
Yes it can and I agree that's a better place for it.  I'll move it.